### PR TITLE
Use numpy.f2py to build via setup.py/pip

### LINKED ISSUE #12: conda recipe for pysolid
Open an issue here to keep track of things for pysolid, so that https://github.com/insarlab/MintPy/issues/648 could be focused on mintpy itself.

@pbrotoisworo are you able to use pysolid on Windows? If so, how do you install it, especially how do you compile the Fortran code?

### PR DESCRIPTION
**Pull request by jhkennedy**
The ultimate goal of this PR is make windows builds via conda-forge (and therefore allow MintPy on windows via conda-forge insarlab/MintPy#648 ). 

It uses the `numpy.f2py` package to provide compilation of the fortran sources at install time via setuptools/pip

```
 python -m pip install git+https://github.com/insarlab/PySolid.git 
 ```

which means:
* pre-compiled versions are no longer needed
* infrastructure for generating pre-compiled versions is no longer needed
* releasing new version should be much simpler

Fixes #12 , #9

### CODE REVIEW

**Comment by jhkennedy** [comment]
I've opened a throw away PR to the PySolid feedstock showing this new build methodology works for Linux, Windows, and Mac conda-forge packaging:
https://github.com/conda-forge/pysolid-feedstock/pull/3

File: setup.py [hunk 1]
```diff
@@ -3,11 +3,15 @@
...
+import setuptools
+from numpy.distutils.core import setup, Extension
+
+# specify fortran extensions to build with numpy.f2py
+solid_ext = Extension(name='pysolid.solid', sources=['pysolid/solid.for'])
 
 setup(
     name='pysolid',
-    version='0.1.2',
+    version='0.2.0',
```
Comment (jhkennedy) [inline setup.py:14]: We'll want to cut a new version for conda-forge, so I've bumped the version number here. Happy to back this out, or change it to a different number.

File: setup.py [hunk 2]
```diff
@@ -3,11 +3,15 @@
 
 
 # always prefer setuptools over distutils
-from setuptools import setup, find_packages
+import setuptools
+from numpy.distutils.core import setup, Extension
```
Comment (jhkennedy) [inline setup.py:7]: importing `setuptools` before `numpy` will insure `numpy.distutils.core` uses modern setuptools methods over distutils methods.

File: setup.py (hunk 1 above)
Comment (yunjunz) [reply setup.py:14]: Good call. Bumping to version `0.2.0` sounds good to me too.

**Comment by yunjunz** [comment]
This is great @jhkennedy!! I tried to follow your note to compile from the source using conda, and the installation seems to work fine. 

There is one issue while running the test commands, as below:

```bash
(test) yunjunz:~/tools/PySolid>$ python -c "import pysolid; print(pysolid.__version__)"
Traceback (most recent call last):
  File "/Users/yunjunz/tools/PySolid/pysolid/grid.py", line 20, in <module>
    from .solid import solid_grid
ModuleNotFoundError: No module named 'pysolid.solid'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/yunjunz/tools/PySolid/pysolid/__init__.py", line 1, in <module>
    from .grid import *
  File "/Users/yunjunz/tools/PySolid/pysolid/grid.py", line 25, in <module>
    raise ImportError(msg)
ImportError: Cannot import name 'solid' from 'pysolid'!
    Maybe solid.for is NOT compiled yet.
    Check instruction at: https://github.com/insarlab/PySolid.
```

I believe this is because both `grid.py` and `point.py` use the local import of the `pysolid.solid` module. I am not sure if this is related to the `libraries` in setup.py. Any idea?

**Comment by yunjunz** [comment]
I am wondering did **the-empty-solid.txt-file** issue (https://github.com/insarlab/PySolid/issues/15) occurred to you for tests of `pysolid.grid` on Windows OS?

**Comment by jhkennedy** [comment]
@yunjunz huh, I can recreate the `ImportError: Cannot import name 'solid' from 'pysolid'!`
 when I install via `pip` without a `-e`, but `solid` is in the installed directory -- wonder why CircleCI or conda-forge isn't having the same problem? I'll see if I can figure out what's going on.

**Comment by yunjunz** [comment]
FYI, if I change the import inside grid.py and point.py from:

```pyrhon
from .solid import solid_grid
```

to 

```python
from pysolid.solid import solid_grid
```

it works fine.

**Comment by jhkennedy** [comment]
@yunjunz yeah if I `cd ..` (out of the PySolid repo) it also works -- I think it's weird python shananigans picking up the local repository and not having the build `.so` inside it (like it does with an editable install). 

I'll change the imports and poke at it a bit more

**Comment by jhkennedy** [comment]
@yunjunz I switch the relative imports to absolute, and fixed a little of the README instructions. *Some* people may still see the same import error if they pip install and then try to run/import pysolid from the the repository root, which is why Python tends to recomend a `src` layout (e.g., [here](https://packaging.python.org/tutorials/packaging-projects/#packaging-python-projects) and [here](https://py-pkgs.org/04-package-structure.html#the-source-layout)). If you're working in the repository, an editable (`-e`) should always work fine and is better for development anyways.

**Comment by yunjunz** [comment]
Thank you @jhkennedy for the fix! The test passed on my macOS.

**Comment by jhkennedy** [comment]
@yunjunz I haven't hand a chance to look at #15 (tests work on linux, focused on the build working on windows), but can try that out as well (will just need to fire up my old desktop).

**Review by yunjunz — APPROVED** [review]
The PR looks all good to me. Thank you @jhkennedy !!!

The `src` structure sounds like a good idea to me. Please feel free to go for it here or in another PR.

As for the testing on windows, it will be great if you could take a look (I don't have access to Windows OS currently). There is no rush, so please take your time.

**Pull request merged**